### PR TITLE
[many ports] Feature 'Introspection' support

### PR DESCRIPTION
**Pull request by mkhon**
Add introspection feature (requires #22199)

- #### What does your PR fix?  

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@mkhon, thanks for your PR! Could you please take a look at the CI testing failure?

atk/gdk-pixbuf with x64-windows, x64-osx, x64-linux:
```
..\src\atk-2-09b87762bd.clean\atk\meson.build:138:2: ERROR: Dependency "gobject-introspection-1.0" not found, tried pkgconfig and cmake
```
pango with x64-windows, x64-osx, x64-linux:
```
Run-time dependency gobject-introspection-1.0 found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency gobject-introspection-1.0

..\src\40a4d027da-8df6f72e5e.clean\pango\meson.build:148:2: ERROR: Automatic wrap-based subproject downloading is disabled
```

graphene with x64-windows:
```
..\src\atk-2-09b87762bd.clean\atk\meson.build:138:2: ERROR: Dependency "gobject-introspection-1.0" not found, tried pkgconfig and cmake
```

**Comment by mkhon** [comment]
This PR depends on #22199 as mentioned in the description

**Comment by PhoebeHui** [comment]
I see, thanks for your response!

**Comment by mkhon** [comment]
Can we remove requires:author-response tag from this one?

**Comment by mkhon** [comment]
@PhoebeHui #22199 has been merged so you can now test this PR

**Comment by PhoebeHui** [comment]
@mkhon, could you merge from master and resolve conflicts? I have no permission to push the changes to this PR.

**Comment by mkhon** [comment]
@PhoebeHui I rebased to the latest master and resolved the conflicts

**Comment by mkhon** [comment]
Looks like latest arm64_windows build failed because of vcpkg bug:

```
Error: vcpkg has crashed; no additional details are available.
```

**Comment by PhoebeHui** [comment]
> Looks like latest arm64_windows build failed because of vcpkg bug:

Yes, it has been fixed in vcpkg-tool repo, and waiting for a new release, please ignore it now.

**Review by PhoebeHui — DISMISSED** [review]

**Comment by ras0219-msft** [comment]
This seems like a maintenance nightmare if it affects every port that touches gobject.

A few questions to help us understand what the problem space is here:
1. Is introspection only supported for meson consumers?
2. Will every consumer of gobject potentially need a flag for this?
3. Does this make sense as a "global" setting? Does it make sense to only enable introspection for _some_ libraries but not others?
4. Is this required to happen as part of the library's build or can it be done separately? For example, could `gtk` generate the introspection data for all of its dependencies?

File: ports/atk/portfile.cmake [hunk 1]
```diff
@@ -14,15 +14,44 @@ vcpkg_extract_source_archive_ex(
...
+        g-ir-scanner='${GIR_TOOL_DIR}/tools/gobject-introspection/g-ir-scanner'
+    ADDITIONAL_CROSS_BINARIES
+        glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'
+        glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'
+        g-ir-compiler='${CURRENT_HOST_INSTALLED_DIR}/tools/gobject-introspection/g-ir-compiler${VCPKG_HOST_EXECUTABLE_SUFFIX}'
+        g-ir-scanner='${GIR_TOOL_DIR}/tools/gobject-introspection/g-ir-scanner'
 )
-vcpkg_install_meson()
+
+vcpkg_host_path_list(APPEND ENV{PKG_CONFIG_PATH} "${CURRENT_INSTALLED_DIR}/lib/pkgconfig")
```
Comment (Neumann-A) [inline ports/atk/portfile.cmake:53]: This looks dangerous. Why is this needed?

**Comment by mkhon** [comment]
> This seems like a maintenance nightmare if it affects every port that touches gobject.
> 
> A few questions to help us understand what the problem space is here:
> 
> 1. Is introspection only supported for meson consumers?

I think not

> 2. Will every consumer of gobject potentially need a flag for this?

Linux packages are usually built with introspection enabled by default. I can modify the PR if the common sense is to do the same

> 3. Does this make sense as a "global" setting? Does it make sense to only enable introspection for _some_ libraries but not others?

I think it makes sense to always build with introspection enabled

> 4. Is this required to happen as part of the library's build or can it be done separately? For example, could `gtk` generate the introspection data for all of its dependencies?

Yes and no.

gtk & friends are packaged so that each component builds its own introspection data (because gtk is theoretically not the only consumer of this). And every gtk package has a knob to enable/disable building introspection. Linux packages tend to not invent the wheel and just build the packages as intended by software authors.

File: ports/atk/portfile.cmake (hunk 1 above)
Comment (mkhon) [reply ports/atk/portfile.cmake]: This was a left-over from an attempt to fix x64-windows-release build. I will submit a separate PR about this (requires more work)

**Comment by JackBoosY** [comment]
Can you please resolve the file conflicts?

**Comment by mkhon** [comment]
@JackBoosY I rebased to latest master. Can we get this merged?

**Comment by JackBoosY** [comment]
Seems there are some file conflicts, can you please resolve that first?

Thanks.

**Comment by JackBoosY** [comment]
@ras0219-msft ping for reply.

**Comment by JackBoosY** [comment]
Can you please merge to master first?

**Comment by JackBoosY** [comment]
Convert this PR to draft since there is no progress.

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.
